src/background/index.js

```diff
@@ -1,30 +1,58 @@
-import { TEXT_SCAN } from '../extensionMessageTypes';
+import { LOOKUP_PLACE, SCAN_PARAGRAPH } from '../extensionMessageTypes';
 
-function selectionTextListener(info) {
+function lookupPlaceListener(info) {
     chrome.tabs.query(
         {
             active: true,
             currentWindow: true,
         },
         (tabs) => {
             chrome.tabs.sendMessage(tabs[0].id, {
-                message: TEXT_SCAN,
+                message: LOOKUP_PLACE,
+                data: info.selectionText,
+            }); // Handle The Response
+        },
+    );
+}
+
+function scanParagraphListener(info) {
+    chrome.tabs.query(
+        {
+            active: true,
+            currentWindow: true,
+        },
+        (tabs) => {
+            chrome.tabs.sendMessage(tabs[0].id, {
+                message: SCAN_PARAGRAPH,
                 data: info.selectionText,
             }); // Handle The Response
         },
     );
 }
 
 const listenerFunctionMap = {
-    selectionWanderland: selectionTextListener,
+    lookupPlaceWanderland: lookupPlaceListener,
+    scanParagraphWanderland: scanParagraphListener,
 };
 
 chrome.runtime.onInstalled.addListener(() => {
     chrome.contextMenus.create({
-        id: 'selectionWanderland',
+        id: 'wanderlandContextMenu',
+        title: 'Wanderland',
+        contexts: ['selection'],
+    });
+    chrome.contextMenus.create({
+        id: 'lookupPlaceWanderland',
+        parentId: 'wanderlandContextMenu',
         title: 'Lookup Place',
         contexts: ['selection'],
     });
+    chrome.contextMenus.create({
+        id: 'scanParagraphWanderland',
+        parentId: 'wanderlandContextMenu',
+        title: 'Scan Paragraph',
```

Comment:
That's a good point. I moved everything to a constants.js file for now, from which we can import stuff. As it's a basic import, it can be changed whenever needed.